[BUG] Error messages appear in a clean build

Doing a clean build will result in multiple `[ERROR]` log messages being emitted.  The errors are benign in practice, and the build succeeds.  It would be nice to not emit unnecessary error messages during the build to avoid the user thinking the build is incorrect/invalid in some way.

Sample error message:
```
[ERROR]  [loadfile] /home/jlowe/src/spark-rapids/sql-plugin/target/spark341/classes/META-INF/services/com.nvidia.spark.rapids.SparkShimServiceProvider doesn't exist
[...]
[INFO]     [unzip] Expanding: /home/jlowe/src/spark-rapids/aggregator/target/spark341/rapids-4-spark-aggregator_2.12-23.12.0-SNAPSHOT-shaded.jar into /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes
[INFO]     [mkdir] Created dir: /home/jlowe/src/spark-rapids/aggregator/target/spark341/old-classes
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: com
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: META-INF
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: org
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: rapids
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: rapids4spark-private-version-info.properties
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: rapids4spark-version-info.properties
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: spark-rapids-extra-configs-classes
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: spark-rapids-extra-plugins
[INFO]      [exec] Only in /home/jlowe/src/spark-rapids/aggregator/target/spark341/new-classes: spark-rapids-extra-startup-configs-classes
[ERROR]      [exec] Result: 1
```

--- a/aggregator/pom.xml
+++ b/aggregator/pom.xml
@@ -156,17 +156,23 @@
                                 <mkdir dir="${newClassesDir}"/>
                                 <delete dir="${newClassesDir}"/>
                                 <unzip src="${realAggJar}" dest="${newClassesDir}"/>
-                                <mkdir dir="${oldClassesDir}"/>
-
-                                <exec executable="diff"
-                                      resultproperty="diff.result"
-                                      discardOutput="${nonfail.errors.quiet}"
-                                      discardError="${nonfail.errors.quiet}">
-                                    <arg value="-q"/>
-                                    <arg value="-r"/>
-                                    <arg value="${oldClassesDir}"/>
-                                    <arg value="${newClassesDir}"/>
-                                </exec>
+                                <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                                    <available file="${oldClassesDir}" type="dir"/>
+                                    <then>
+                                        <exec executable="diff"
+                                            resultproperty="diff.result"
+                                            discardOutput="${nonfail.errors.quiet}"
+                                            discardError="${nonfail.errors.quiet}">
+                                            <arg value="-q"/>
+                                            <arg value="-r"/>
+                                            <arg value="${oldClassesDir}"/>
+                                            <arg value="${newClassesDir}"/>
+                                        </exec>
+                                    </then>
+                                    <else>
+                                        <echo>Clean build? Skipping diff because ${oldClassesDir} does not exist</echo>
+                                    </else>
+                                </ac:if>
 
                                 <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                                     <equals arg1="0" arg2="${diff.result}"/>

--- a/aggregator/pom.xml
+++ b/aggregator/pom.xml
@@ -159,7 +159,9 @@
                                 <mkdir dir="${oldClassesDir}"/>
 
                                 <exec executable="diff"
-                                      resultproperty="diff.result">
+                                      resultproperty="diff.result"
+                                      discardOutput="${nonfail.errors.quiet}"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="-q"/>
                                     <arg value="-r"/>
                                     <arg value="${oldClassesDir}"/>

--- a/pom.xml
+++ b/pom.xml
@@ -820,7 +820,7 @@
         <bloop.installPhase>install</bloop.installPhase>
         <bloop.configDirectory>${spark.rapids.source.basedir}/.bloop</bloop.configDirectory>
         <build.info.path>${project.build.outputDirectory}/rapids4spark-version-info.properties</build.info.path>
-        <nonfail.errors.quiet>true</nonfail.errors.quiet>
+        <nonfail.errors.quiet>false</nonfail.errors.quiet>
     </properties>
 
     <dependencyManagement>
@@ -985,13 +985,22 @@
                         <configuration>
                             <!-- Execute the shell script to generate the plugin build information. -->
                             <target name="build-info">
+                                <taskdef resource="net/sf/antcontrib/antcontrib.properties"/>
                                 <exec executable="git"
                                       outputproperty="git.head.revision"
                                       discardError="${nonfail.errors.quiet}">
                                     <arg value="rev-parse"/>
                                     <arg value="HEAD"/>
                                 </exec>
-                                <property file="${build.info.path}" prefix="saved.build-info"/>
+                                <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                                    <available file="${build.info.path}" type="file"/>
+                                    <then>
+                                        <property file="${build.info.path}" prefix="saved.build-info"/>
+                                    </then>
+                                    <else>
+                                        <property name="saved.build-info.revision" value="N/A"/>
+                                    </else>
+                                </ac:if>
                                 <echo>
 Comparing git revisions:
     previous=${saved.build-info.revision}

--- a/pom.xml
+++ b/pom.xml
@@ -820,6 +820,7 @@
         <bloop.installPhase>install</bloop.installPhase>
         <bloop.configDirectory>${spark.rapids.source.basedir}/.bloop</bloop.configDirectory>
         <build.info.path>${project.build.outputDirectory}/rapids4spark-version-info.properties</build.info.path>
+        <nonfail.errors.quiet>true</nonfail.errors.quiet>
     </properties>
 
     <dependencyManagement>
@@ -984,7 +985,9 @@
                         <configuration>
                             <!-- Execute the shell script to generate the plugin build information. -->
                             <target name="build-info">
-                                <exec executable="git" outputproperty="git.head.revision">
+                                <exec executable="git"
+                                      outputproperty="git.head.revision"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="rev-parse"/>
                                     <arg value="HEAD"/>
                                 </exec>
@@ -1471,7 +1474,9 @@ This will force full Scala code rebuild in downstream modules.
                                 </pathconvert>
                                 <echo>Cleaning build directories of all modules ${target.dirs.str}</echo>
                                 <!-- workaround ant delete does not work with dirset -->
-                                <exec dir="${project.basedir}" executable="rm">
+                                <exec dir="${project.basedir}"
+                                      executable="rm"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="-rf"/>
                                     <arg line="${target.dirs.str}"/>
                                 </exec>

--- a/scala2.13/aggregator/pom.xml
+++ b/scala2.13/aggregator/pom.xml
@@ -156,17 +156,23 @@
                                 <mkdir dir="${newClassesDir}"/>
                                 <delete dir="${newClassesDir}"/>
                                 <unzip src="${realAggJar}" dest="${newClassesDir}"/>
-                                <mkdir dir="${oldClassesDir}"/>
-
-                                <exec executable="diff"
-                                      resultproperty="diff.result"
-                                      discardOutput="${nonfail.errors.quiet}"
-                                      discardError="${nonfail.errors.quiet}">
-                                    <arg value="-q"/>
-                                    <arg value="-r"/>
-                                    <arg value="${oldClassesDir}"/>
-                                    <arg value="${newClassesDir}"/>
-                                </exec>
+                                <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                                    <available file="${oldClassesDir}" type="dir"/>
+                                    <then>
+                                        <exec executable="diff"
+                                            resultproperty="diff.result"
+                                            discardOutput="${nonfail.errors.quiet}"
+                                            discardError="${nonfail.errors.quiet}">
+                                            <arg value="-q"/>
+                                            <arg value="-r"/>
+                                            <arg value="${oldClassesDir}"/>
+                                            <arg value="${newClassesDir}"/>
+                                        </exec>
+                                    </then>
+                                    <else>
+                                        <echo>Clean build? Skipping diff because ${oldClassesDir} does not exist</echo>
+                                    </else>
+                                </ac:if>
 
                                 <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                                     <equals arg1="0" arg2="${diff.result}"/>

--- a/scala2.13/aggregator/pom.xml
+++ b/scala2.13/aggregator/pom.xml
@@ -159,7 +159,9 @@
                                 <mkdir dir="${oldClassesDir}"/>
 
                                 <exec executable="diff"
-                                      resultproperty="diff.result">
+                                      resultproperty="diff.result"
+                                      discardOutput="${nonfail.errors.quiet}"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="-q"/>
                                     <arg value="-r"/>
                                     <arg value="${oldClassesDir}"/>

--- a/scala2.13/pom.xml
+++ b/scala2.13/pom.xml
@@ -820,7 +820,7 @@
         <bloop.installPhase>install</bloop.installPhase>
         <bloop.configDirectory>${spark.rapids.source.basedir}/.bloop</bloop.configDirectory>
         <build.info.path>${project.build.outputDirectory}/rapids4spark-version-info.properties</build.info.path>
-        <nonfail.errors.quiet>true</nonfail.errors.quiet>
+        <nonfail.errors.quiet>false</nonfail.errors.quiet>
     </properties>
 
     <dependencyManagement>
@@ -985,13 +985,22 @@
                         <configuration>
                             <!-- Execute the shell script to generate the plugin build information. -->
                             <target name="build-info">
+                                <taskdef resource="net/sf/antcontrib/antcontrib.properties"/>
                                 <exec executable="git"
                                       outputproperty="git.head.revision"
                                       discardError="${nonfail.errors.quiet}">
                                     <arg value="rev-parse"/>
                                     <arg value="HEAD"/>
                                 </exec>
-                                <property file="${build.info.path}" prefix="saved.build-info"/>
+                                <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                                    <available file="${build.info.path}" type="file"/>
+                                    <then>
+                                        <property file="${build.info.path}" prefix="saved.build-info"/>
+                                    </then>
+                                    <else>
+                                        <property name="saved.build-info.revision" value="N/A"/>
+                                    </else>
+                                </ac:if>
                                 <echo>
 Comparing git revisions:
     previous=${saved.build-info.revision}

--- a/scala2.13/pom.xml
+++ b/scala2.13/pom.xml
@@ -994,7 +994,7 @@
                                 <property file="${build.info.path}" prefix="saved.build-info"/>
                                 <echo>
 Comparing git revisions:
-    previous=${saved.build-info.revision} from ${build.info.path}
+    previous=${saved.build-info.revision}
      current=${git.head.revision}
                                 </echo>
                                 <taskdef resource="net/sf/antcontrib/antcontrib.properties"/>

--- a/scala2.13/pom.xml
+++ b/scala2.13/pom.xml
@@ -820,6 +820,7 @@
         <bloop.installPhase>install</bloop.installPhase>
         <bloop.configDirectory>${spark.rapids.source.basedir}/.bloop</bloop.configDirectory>
         <build.info.path>${project.build.outputDirectory}/rapids4spark-version-info.properties</build.info.path>
+        <nonfail.errors.quiet>true</nonfail.errors.quiet>
     </properties>
 
     <dependencyManagement>
@@ -984,7 +985,9 @@
                         <configuration>
                             <!-- Execute the shell script to generate the plugin build information. -->
                             <target name="build-info">
-                                <exec executable="git" outputproperty="git.head.revision">
+                                <exec executable="git"
+                                      outputproperty="git.head.revision"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="rev-parse"/>
                                     <arg value="HEAD"/>
                                 </exec>
@@ -1471,7 +1474,9 @@ This will force full Scala code rebuild in downstream modules.
                                 </pathconvert>
                                 <echo>Cleaning build directories of all modules ${target.dirs.str}</echo>
                                 <!-- workaround ant delete does not work with dirset -->
-                                <exec dir="${project.basedir}" executable="rm">
+                                <exec dir="${project.basedir}"
+                                      executable="rm"
+                                      discardError="${nonfail.errors.quiet}">
                                     <arg value="-rf"/>
                                     <arg line="${target.dirs.str}"/>
                                 </exec>

--- a/scala2.13/pom.xml
+++ b/scala2.13/pom.xml
@@ -994,7 +994,7 @@
                                 <property file="${build.info.path}" prefix="saved.build-info"/>
                                 <echo>
 Comparing git revisions:
-    previous=${saved.build-info.revision}
+    previous=${saved.build-info.revision} from ${build.info.path}
      current=${git.head.revision}
                                 </echo>
                                 <taskdef resource="net/sf/antcontrib/antcontrib.properties"/>

--- a/scala2.13/sql-plugin/pom.xml
+++ b/scala2.13/sql-plugin/pom.xml
@@ -179,10 +179,18 @@
                             value="${servicesDir}/com.nvidia.spark.rapids.SparkShimServiceProvider"/>
                         <property name="shimServiceClass"
                             value="com.nvidia.spark.rapids.shims.${spark.version.classifier}.SparkShimServiceProvider"/>
-                        <loadfile property="currentServiceClass"
-                                  srcFile="${serviceFile}"
-                                  failonerror="false"
-                                  quiet="${nonfail.errors.quiet}"/>
+                        <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                            <available file="${serviceFile}" type="file"/>
+                            <then>
+                                <loadfile property="currentServiceClass"
+                                    srcFile="${serviceFile}"
+                                    failonerror="false"
+                                    quiet="${nonfail.errors.quiet}"/>
+                            </then>
+                            <else>
+                                <property name="currentServiceClass" value="N/A"/>
+                            </else>
+                        </ac:if>
                         <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                             <equals arg1="${currentServiceClass}" arg2="${shimServiceClass}${line.separator}"/>
                             <then>

--- a/scala2.13/sql-plugin/pom.xml
+++ b/scala2.13/sql-plugin/pom.xml
@@ -179,7 +179,10 @@
                             value="${servicesDir}/com.nvidia.spark.rapids.SparkShimServiceProvider"/>
                         <property name="shimServiceClass"
                             value="com.nvidia.spark.rapids.shims.${spark.version.classifier}.SparkShimServiceProvider"/>
-                        <loadfile property="currentServiceClass" srcFile="${serviceFile}" failonerror="false"/>
+                        <loadfile property="currentServiceClass"
+                                  srcFile="${serviceFile}"
+                                  failonerror="false"
+                                  quiet="${nonfail.errors.quiet}"/>
                         <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                             <equals arg1="${currentServiceClass}" arg2="${shimServiceClass}${line.separator}"/>
                             <then>

--- a/sql-plugin/pom.xml
+++ b/sql-plugin/pom.xml
@@ -179,10 +179,18 @@
                             value="${servicesDir}/com.nvidia.spark.rapids.SparkShimServiceProvider"/>
                         <property name="shimServiceClass"
                             value="com.nvidia.spark.rapids.shims.${spark.version.classifier}.SparkShimServiceProvider"/>
-                        <loadfile property="currentServiceClass"
-                                  srcFile="${serviceFile}"
-                                  failonerror="false"
-                                  quiet="${nonfail.errors.quiet}"/>
+                        <ac:if xmlns:ac="antlib:net.sf.antcontrib">
+                            <available file="${serviceFile}" type="file"/>
+                            <then>
+                                <loadfile property="currentServiceClass"
+                                    srcFile="${serviceFile}"
+                                    failonerror="false"
+                                    quiet="${nonfail.errors.quiet}"/>
+                            </then>
+                            <else>
+                                <property name="currentServiceClass" value="N/A"/>
+                            </else>
+                        </ac:if>
                         <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                             <equals arg1="${currentServiceClass}" arg2="${shimServiceClass}${line.separator}"/>
                             <then>

--- a/sql-plugin/pom.xml
+++ b/sql-plugin/pom.xml
@@ -179,7 +179,10 @@
                             value="${servicesDir}/com.nvidia.spark.rapids.SparkShimServiceProvider"/>
                         <property name="shimServiceClass"
                             value="com.nvidia.spark.rapids.shims.${spark.version.classifier}.SparkShimServiceProvider"/>
-                        <loadfile property="currentServiceClass" srcFile="${serviceFile}" failonerror="false"/>
+                        <loadfile property="currentServiceClass"
+                                  srcFile="${serviceFile}"
+                                  failonerror="false"
+                                  quiet="${nonfail.errors.quiet}"/>
                         <ac:if xmlns:ac="antlib:net.sf.antcontrib">
                             <equals arg1="${currentServiceClass}" arg2="${shimServiceClass}${line.separator}"/>
                             <then>

